src/overlay/OverlayManagerImpl.h

```diff
@@ -184,6 +187,18 @@ class OverlayManagerImpl : public OverlayManager
     RandomEvictionCache<Hash, std::weak_ptr<CapacityTrackedMessage>>
         mScheduledMessages;
 
+    // Ledger snapshot used for background transaction validation. Must be
+    // updated (replaced) after each new ledger via the `updateSnapshots`
+    // function.
+    std::unique_ptr<ExtendedLedgerSnapshot const> mLedgerSnapshot;
```

Comment:
It's safer to create a new LedgerSnapshot every time you need to use it to avoid caching inconsistencies.